Sources/SnapshotTesting/Snapshotting/UIImage.swift

```diff
@@ -14,21 +14,30 @@ extension Diffing where Value == UIImage {
     return Diffing(
       toData: { $0.pngData() ?? emptyImage().pngData()! },
       fromData: { UIImage(data: $0, scale: UIScreen.main.scale)! }
-    ) { old, new in
+    ) { old, new -> (String, [SnapshotArtifact])? in
       guard !compare(old, new, precision: precision) else { return nil }
       let difference = SnapshotTesting.diff(old, new)
       let message = new.size == old.size
         ? "Newly-taken snapshot does not match reference."
         : "Newly-taken snapshot@\(new.size) does not match reference@\(old.size)."
-      let oldAttachment = XCTAttachment(image: old)
-      oldAttachment.name = "reference"
-      let newAttachment = XCTAttachment(image: new)
-      newAttachment.name = "failure"
-      let differenceAttachment = XCTAttachment(image: difference)
-      differenceAttachment.name = "difference"
-      return (
+        let oldArtifact = SnapshotArtifact(
+            data: old.pngData()!,
```

Comment:
do we need force unwrap here? (and same for lines below)

Reply:
This is taken from the original PR on the library, approved by the maintainers. Given that this is a test, I would say the force unwrap is worth it 🧑‍🎓